projects/transformers/run.py

```diff
@@ -643,21 +824,32 @@ def run_finetuning_multiple_tasks(
             set_seed(training_args.seed)
 
             if model_args.hp_num_trials > 1:
-                eval_results = run_finetuning_single_task_with_hp_search(
-                    model_args,
-                    data_args,
-                    training_args,
-                    last_checkpoint=last_checkpoint
-                )
+                if data_args.task_name == "squad":
+                    print("Hp search for squad is not yet implemented")
```

Comment:
Would it be hard to extend the existing hp search method for squad? When I initially coded I planned to make it generic enough to handle hp search for any finetuning task - if that is not the case, then it is a fail of the initial design, and the todo should be to fix that hp search instead of making a new one.

Reply:
It's not a huge leap, I just didn't have time for it yet so I put this in to clarify that I haven't gotten to that yet. The main lift is the way squad datasets and models are initialized. First, squad datasets are initialized with separate code currently. I know you commented above that you don't want that to be the case, but it takes a lot of code to preprocess squad, so either we can call the preprocessing code separately in the hp search function, or we can make another big lift to merge squad code into finetuning code.  Second, as you may recall, we have to define a special model init and pass that to the trainer.  This is a small modification.